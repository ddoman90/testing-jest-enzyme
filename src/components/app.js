import React from 'react';
import CommentBox from './commentBox';
import CommentList from './commentList';

export default props => {
	return (
		<div className="comment-box">
			<h4>Add comment</h4>
			<CommentBox />
			<CommentList />
		</div>
	);
};
