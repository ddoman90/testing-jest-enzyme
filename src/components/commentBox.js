import React, { Component } from 'react';
import { saveComment } from '../actions';
import { connect } from 'react-redux';

const INITIAL_STATE = {
	comment: ''
};

export class CommentBox extends Component {
	constructor(props) {
		super(props);

		this.state = INITIAL_STATE;
	}
	onChange({ target }) {
		this.setState({
			comment: target.value
		});
	}

	onSubmit(event) {
		event.preventDefault();
		this.props.saveComment(this.state.comment);
		this.setState(INITIAL_STATE);
	}

	render() {
		return (
			<form className="comment_box" onSubmit={this.onSubmit.bind(this)}>
				<textarea
					className="comment_text"
					name="comment"
					value={this.state.comment}
					onChange={this.onChange.bind(this)}
				/>
				<div>
					<button action="submit">Submit</button>
				</div>
			</form>
		);
	}
}

export default connect(null, { saveComment })(CommentBox);
