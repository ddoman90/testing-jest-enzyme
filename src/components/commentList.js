import React, { Component } from 'react';
import { connect } from 'react-redux';

export class CommentList extends Component {
	render() {
		const comments = this.props.comments || [];
		return (
			<div className="comment_list">
				{comments.map(comment => <li key={comment}>{comment}</li>)}
			</div>
		);
	}
}

export default connect(({ comments }) => {
	return { comments };
})(CommentList);
