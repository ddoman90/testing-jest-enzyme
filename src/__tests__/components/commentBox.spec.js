import React from 'react';
import { shallow } from 'enzyme';
// import the dumb component for unit test
import { CommentBox } from '../../components/commentBox';
import renderer from 'react-test-renderer';

describe('CommentBox', () => {
	let component;

	beforeEach(() => {
		component = shallow(<CommentBox />);
	});

	it('renders with the correct class', () => {
		expect(component.is('.comment_box')).toBe(true);
	});

	it('renders correctly', () => {
		const tree = renderer.create(<CommentBox />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders a textarea', () => {
		expect(component.find('textarea').length).toBe(1);
	});

	it('renders a button', () => {
		expect(component.find('button').length).toBe(1);
	});

	describe('entering some text', () => {
		let textarea;
		beforeEach(() => {
			textarea = component.find('textarea');
			textarea.simulate('change', { target: { value: 'new comment' } });
		});

		it('shows the entered text', () => {
			expect(component.find('textarea').prop('value')).toBe(
				'new comment'
			);
		});

		it('shows empty field after submission', () => {
			component.simulate('submit', { preventDefault: function() {} });
			expect(component.find('textarea').prop('value')).toBe('');
		});
	});
});
