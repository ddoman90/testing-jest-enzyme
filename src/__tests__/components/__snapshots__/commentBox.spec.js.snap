// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentBox renders correctly 1`] = `
<form
  className="comment_box"
  onSubmit={[Function]}
>
  <textarea
    className="comment_text"
    name="comment"
    onChange={[Function]}
    value=""
  />
  <div>
    <button
      action="submit"
    >
      Submit
    </button>
  </div>
</form>
`;
